executor/aggfuncs/func_json_objectagg.go

```diff
@@ -109,6 +111,21 @@ func (e *jsonObjectAgg) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup
 		keyString = stringutil.Copy(keyString)
 
 		realVal := value.Clone().GetValue()
+		// Special handling for varbinary
+		if e.args[1].GetType().Tp == mysql.TypeVarchar && mysql.HasBinaryFlag(e.args[1].GetType().Flag) {
```

Comment:
Looks like it's not only for json_objectagg(). json_object() also has same behavior.